### PR TITLE
test(client): Sanitize error snapshots

### LINKED ISSUE #14321: Sanitize error snapshots in new test setup
Our error snapshots contain file paths and line numbers of the offending code. This causes the need to update snapshots every time we add or remove a line below, which causes additional noise during code reviews. We should sanitize line/column information to make snapshots more resilient. For example, we could sanitize them to look like: 

```
Invalid \`prisma.user.create()\` invocation in
/client/tests/functional/interactive-transactions/tests.ts:XX:XX
  XX  */
  XX testIf(getClientEngineType() === ClientEngineType.Library)('batching rollback', async () => {
  XX   const result = prisma.$transaction([
→ XX    prisma.user.create(
```

### PR DESCRIPTION
**Pull request by SevInf**
Introduce custom jest matchers that sanitize output of
`printError` function. I opted for custom matcher instead of
adding this functionality to default snapshot serializer because there
might be some cases, where we actually want to see the line numbers in
the snapshots (#14265 is one example). This won't be possible if line
numbers are always sanitized.

Close #14321

### CODE REVIEW

**Review by danstarns — COMMENTED** [review]
Should this wait until your referred PR has been merged so we know we have coverage for the line numbers?

**Comment by SevInf** [comment]
> Should this wait until your referred PR has been merged so we know we have coverage for the line numbers?

That's a good idea, i'll try to find time for reviving that one soon.

**Review by danstarns — APPROVED** [review]
Now that you merged [the refactor of the print stack function](https://github.com/prisma/prisma/pull/14265), with its unit tests, the changes here are fine to merge. Nice work - this is soo much more manageable.

**Pull request merged**
